pkg/arvo/app/chat-hook.hoon

```diff
@@ -79,7 +77,7 @@
           ^-  (list (list card))
           %+  turn  ~(tap in keys)  generate-cards
         ==
-    [~ this(state upgraded-state)]
+    [loaded-cards.upgraded-state this(state upgraded-state)]
```

Comment:
If the hook upgrades first, the store moves in this are going to crash, right?

I think the thing to do here is just track whether one condition (of "store upgraded" and "hook upgraded") has already been met in a flag. Then, if we're trying to set the flag to true, but it already is, kick off the upgrade logic.